src/uwtools/config/jinja2.py

```diff
@@ -266,6 +269,9 @@ def _deref_render(val: str, context: dict, local: Optional[dict] = None) -> str:
     context = {**(local or {}), **context}
     try:
         rendered = _register_filters(env).from_string(val).render(context)
+        if isinstance(yaml.load(rendered, Loader=uw_yaml_loader()), UWYAMLConvert):
+            _deref_debug("Held", rendered)
+            raise UWConfigRealizeError()
```

Comment:
Here's the crux of the bugfix: If the rendered value turns out to be something (e.g. `!int '{{ foo }}'`) that loads as a `UWYAMLConvert` object, reject the rendered value, and hold the old value, by raising an exception. The value may be rendered by a future iteration of the dereferencing loop when the value it refers to has been reduced to a basic value (a `bool`, `int`, etc.).